test/integration/protocol_integration_test.cc

```diff
@@ -4915,6 +4915,258 @@ TEST_P(ProtocolIntegrationTest, InvalidResponseHeaderNameStreamError) {
   ASSERT_TRUE(fake_upstream_connection_->connected());
 }
 
+// Validate that when allow_multiplexed_upstream_half_close is enabled a request with H/2 or H/3
```

Comment:
I think we should add some explicit half closes to the filter integration test, making sure half close works correctly in the "add body / trailers" case as well as "router filter is not terminal filter" case